examples/frost.c

```diff
@@ -0,0 +1,291 @@
+/***********************************************************************
+ * Copyright (c) 2021-2023  Jesse Posner                               *
+ * Distributed under the MIT software license, see the accompanying    *
+ * file COPYING or https://www.opensource.org/licenses/mit-license.php.*
+ ***********************************************************************/
+
+/**
+ * This file demonstrates how to use the FROST module to create a threshold
+ * signature. Additionally, see the documentation in include/secp256k1_frost.h.
+ */
+
+#include <stdio.h>
+#include <assert.h>
+#include <string.h>
+
+#include <secp256k1.h>
+#include <secp256k1_schnorrsig.h>
+#include <secp256k1_frost.h>
+
+#include "examples_util.h"
+/* Number of public keys involved in creating the aggregate signature */
+#define N_SIGNERS 5
+
+/* Threshold required in creating the aggregate signature */
+#define THRESHOLD 3
+
+struct signer_secrets {
+    secp256k1_keypair keypair;
+    secp256k1_frost_share agg_share;
+    secp256k1_frost_secnonce secnonce;
+    unsigned char seed[32];
+};
+
+struct signer {
+    secp256k1_pubkey pubshare;
+    secp256k1_frost_pubnonce pubnonce;
+    secp256k1_frost_session session;
+    secp256k1_frost_partial_sig partial_sig;
+    secp256k1_pubkey vss_commitment[THRESHOLD];
+    unsigned char vss_hash[32];
+    unsigned char pok[64];
+    unsigned char id[33];
+};
+
+/* Create a key pair and store it in seckey and pubkey */
+int create_keypair(const secp256k1_context* ctx, struct signer_secrets *signer_secrets, struct signer *signer) {
+    unsigned char seckey[32];
+    secp256k1_pubkey pubkey_tmp;
+    size_t size = 33;
+
+    while (1) {
+        if (!fill_random(seckey, sizeof(seckey))) {
+            printf("Failed to generate randomness\n");
+            return 1;
+        }
+        if (secp256k1_keypair_create(ctx, &signer_secrets->keypair, seckey)) {
+            break;
+        }
+    }
+    if (!secp256k1_keypair_pub(ctx, &pubkey_tmp, &signer_secrets->keypair)) {
+        return 0;
+    }
+    if (!secp256k1_ec_pubkey_serialize(ctx, signer->id, &size, &pubkey_tmp, SECP256K1_EC_COMPRESSED)) {
+        return 0;
+    }
+    return 1;
+}
+
+/* Create shares and coefficient commitments */
+int create_shares(const secp256k1_context* ctx, struct signer_secrets *signer_secrets, struct signer *signer, secp256k1_xonly_pubkey *agg_pk) {
+    int i, j;
+    secp256k1_frost_share shares[N_SIGNERS][N_SIGNERS];
+    const secp256k1_pubkey *vss_commitments[N_SIGNERS];
+    const unsigned char *ids[N_SIGNERS];
+
+    for (i = 0; i < N_SIGNERS; i++) {
+        vss_commitments[i] = signer[i].vss_commitment;
+        ids[i] = signer[i].id;
+    }
+
+    for (i = 0; i < N_SIGNERS; i++) {
+        /* Generate a polynomial share for the participants */
+        if (!secp256k1_frost_shares_gen(ctx, shares[i], signer[i].vss_commitment, signer[i].pok, signer_secrets[i].seed, THRESHOLD, N_SIGNERS, ids)) {
```

Comment:
I checked why the valgrind CI jobs fail and it looks like `signer_secrets[i].seed` is never initialized.

Reply:
Ah, good catch. Thanks for tracking that down!